core/zip/src/RZip.cxx

```diff
@@ -79,7 +79,7 @@ unsigned long R__crc32(unsigned long crc, const unsigned char* buf, unsigned int
 void R__zipMultipleAlgorithm(int cxlevel, int *srcsize, char *src, int *tgtsize, char *tgt, int *irep, ROOT::RCompressionSetting::EAlgorithm::EValues compressionAlgorithm)
 {
 
-  if (*srcsize < 1 + HDRSIZE + 1) {
+  if (*tgtsize < 1 + HDRSIZE + 1) {
```

Comment:
This change is semantically fishy ... The code was added with the following log:
```
    Add threshold to avoid compressing small buffers
    
    While tracing CMSSW 25202.0 workflow GEN-SIM step I found that majority
    of compressions were of a single (1) byte. Compressing one byte gives us
    back one byte and another 9 bytes (for header) would be written. Add 10
    byte threshold to avoid compressing small buffers.
```
i.e.  It is really about the `srcsize` being low (side bar: the threshold probably should be higher).  Technically the `tgtsize` could be independently large as at this point it indicates the amount/size of the allocated output buffer.

Reply:
Ok, in that case we want both checks